src/ucp/core/ucp_request.inl

```diff
@@ -114,6 +114,15 @@ ucp_request_complete_stream_recv(ucp_request_t *req,
     ucp_request_complete(req, recv.stream.cb, status, req->recv.stream.length);
 }
 
+static UCS_F_ALWAYS_INLINE int
+ucp_request_can_complete_stream_recv(ucp_request_t *req)
+{
+    return (req->recv.state.offset > 0) &&
+           (ucs_likely(UCP_DT_IS_CONTIG(req->recv.datatype)) ?
+            (req->recv.state.offset %
+             ucp_contig_dt_elem_size(req->recv.datatype) == 0) : 1);
```

Comment:
as @brminich noted, need to avoid the modulo, it's not a cheap operation.. can add 10-30 nsec

Reply:
This check will be called per AM (rdesc), this will affect only when user do a lot of small sends to 1 big recv. I think we can avoid modulo operation even in this situation but not for free...
For now, i can add `req->recv.state.offset == req->recv.length` before `%` to avoid it on full request. What do you think?